packages/web-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { createElement, useRef, LegacyRef } from 'react';
+import classNames from 'classnames';
+import { useStyleProps } from '../../hooks/styleProps';
+import { SpiritDropdownProps } from '../../types';
+import { useDropdown } from './useDropdown';
+import { useDropdownStyleProps } from './useDropdownStyleProps';
+import { useDropdownAriaProps } from './useDropdownAriaProps';
+
+const Dropdown = (props: SpiritDropdownProps) => {
+  const { id = Math.random().toString(36).slice(2, 7), children, renderTrigger, disableAutoClose, ...rest } = props;
+
+  const dropdownRef = useRef();
+  const triggerRef = useRef();
+
+  const { isOpen, toggleHandler } = useDropdown({ dropdownRef, triggerRef, disableAutoClose });
+  const { wrapperClassName, triggerClassName } = useDropdownStyleProps({ isOpen });
+  const { triggerProps } = useDropdownAriaProps({ id, isOpen, toggleHandler });
+
+  const { styleProps: wrapperStyleProps, props: wrapperOtherProps } = useStyleProps({ ...rest });
+  const { styleProps: triggerStyleProps } = useStyleProps({
+    UNSAFE_className: triggerClassName,
+  });
+
+  const triggerRenderHandler = () => {
```

Comment:
I think we should use an implicit return here + ternary operator. Two returns for this simple function is too much, especially without using parentheses in the condition.

Reply:
```
  const triggerRenderHandler = () =>
    renderTrigger
      ? renderTrigger({
          isOpen,
          trigger: {
            ...triggerStyleProps,
            ...triggerProps,
            ref: triggerRef as unknown as LegacyRef<HTMLButtonElement & HTMLAnchorElement>,
          },
        })
      : null;
```